### PR TITLE
feat(dsg): upgrade react-admin to v4

### PR DESCRIPTION
**Pull request by Meyazhagan**
Close: #7395 

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by overbit** [comment]
Thanks @Meyazhagan! we will review it ASAP.

**Comment by Shurtu-gal** [comment]
@overbit should we test if all plugins work as expected after this version bump?

**Comment by overbit** [comment]
@Shurtu-gal yep, that will be necessary. Enterprise users currently have the ability to select the DSG version so they could stay with the current plugin versions but all others will need to have all the latest official plugin versions fully working with react admin 4.

Special attention might be required for the Auth plugins as with the latest react admin, there are changes to the AuthProvider interface and hooks.

@mulygottlieb this needs to be taken into consideration

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/data-service-generator/src/admin/entity/create-field-value.ts
[the remark this answers is not in the document]
Comment (Meyazhagan) [reply packages/data-service-generator/src/admin/entity/create-field-value.ts:1]: yeah sure

File: packages/data-service-generator/src/admin/entity/entity-list-component/entity-list-component.template.tsx [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
+export const ENTITY_LIST = (): React.ReactElement => {
   return (
-    <List
-      {...props}
-      bulkActionButtons={false}
-      title={ENTITY_PLURAL_DISPLAY_NAME}
-      perPage={50}
-      pagination={<Pagination />}
-    >
+    <List>
```
Comment (Shurtu-gal) [inline packages/data-service-generator/src/admin/entity/entity-list-component/entity-list-component.template.tsx:11]: Are the list props no longer needed ?

**Review by Shurtu-gal — CHANGES_REQUESTED** [review]

File: packages/data-service-generator/src/admin/entity/entity-list-component/entity-list-component.template.tsx (hunk 1 above)
Comment (Meyazhagan) [reply packages/data-service-generator/src/admin/entity/entity-list-component/entity-list-component.template.tsx:11]: `bulkActionButtons` is deprecated. we can have rest of props.

Comment (Meyazhagan) [reply packages/data-service-generator/src/admin/entity/entity-list-component/entity-list-component.template.tsx:11]: I saw in pagination component, we extending rowsPerPageOptions, so I revert that change

Comment (Meyazhagan) [reply packages/data-service-generator/src/admin/entity/entity-list-component/entity-list-component.template.tsx:11]: > Are the list props no longer needed ?

yes

**Comment by mulygottlieb** [comment]
@Meyazhagan  @Shurtu-gal @overbit I am sorry, but I'm not sure why we are investing time in this.
We focus on backend services generation, and the react admin UI is a nice-to-have utility. Spending time on this + adding the potential risk of breaking existing plugins and requiring significant regression testing doesn't make sense to me.

**Pull request closed without merging**